Kernel/Interrupts/PCIIRQHandler.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023, Pankaj R <dev@pankajraghav.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/InterruptManagement.h>
+#include <Kernel/Arch/PCIMSI.h>
+#include <Kernel/Debug.h>
+#include <Kernel/Interrupts/PCIIRQHandler.h>
+
+namespace Kernel {
+
+PCIIRQHandler::PCIIRQHandler(PCI::Device& device, u8 irq)
+    : GenericInterruptHandler(irq)
+    , device(device)
+{
+    auto type = device.get_interrupt_type();
+
+    if (type == PCI::InterruptType::PIN)
+        m_responsible_irq_controller = InterruptManagement::the().get_responsible_irq_controller(irq);
+    else
+        m_message_signalled_interrupts = true;
```

Comment:
The variables `m_message_signalled_interrupts` and `m_responsible_irq_controller` duplicate a bit of state. I suggest dropping `m_message_signalled_interrupts` and checking `!m_responsible_irq_controller.is_null()` to see if you can invoke the IRQ controller.